File: packages/rules/test/fixtures/slack/py/missing-signature-verification.negative.py
import os

from flask import Flask, request
from slack_sdk.signature import SignatureVerifier

app = Flask(__name__)
verifier = SignatureVerifier(os.environ["SLACK_SIGNING_SECRET"])


@app.route("/slack/events", methods=["POST"])
def slack_events():
    body = request.get_data().decode("utf-8")
    if not verifier.is_valid_request(body, request.headers):
        return "", 403
    return "", 200

File: packages/rules/test/fixtures/slack/js/missing-signature-verification.negative.ts
import { verifyRequestSignature } from "@slack/events-api";
import express from "express";

const app = express();
app.use(express.raw({ type: "application/json" }));

app.post("/slack/events", async (req, res) => {
  const requestSignature = (req.header("X-Slack-Signature") ?? "") as string;
  const requestTimestamp = Number(req.header("X-Slack-Request-Timestamp") ?? "0");
  const body = (req.body as Buffer).toString("utf8");
  const valid = verifyRequestSignature({
    signingSecret: process.env.SLACK_SIGNING_SECRET ?? "",
    requestSignature,
    requestTimestamp,
    body,
  });
  if (!valid) return res.status(403).end();
  res.status(200).send("");
});

app.listen(3000);
